### PR TITLE
[WIP] on-off bookinfo into ambient

### PR DESCRIPTION
**Pull request by irisdingbj**
**Please provide a description of this PR:**

### CODE REVIEW

File: tests/integration/ambient/main_test.go [hunk 1]
```diff
@@ -88,7 +88,6 @@ func TestMain(m *testing.M) {
 		Setup(istio.Setup(&i, func(ctx resource.Context, cfg *istio.Config) {
 			// can't deploy VMs without eastwest gateway
 			ctx.Settings().SkipVMs()
```
Comment (bleggett) [inline tests/integration/ambient/main_test.go:90]: I'm a bit confused - how does dropping this flag change things, exactly? 

```
	if cfg.EnableCNI {
		args.AppendSet("components.cni.namespace", "kube-system")
		args.AppendSet("components.cni.enabled", "true")
	}
```

how/why does this make bookinfo tests flake, exactly?

We _do_ want cni enabled for ambient?

Comment (irisdingbj) [reply tests/integration/ambient/main_test.go:90]: Just found out in my local running, the istio-cni-node pod has not been running at all.    Tried to remove this line and the istio-cni-node pod is up and running though not sure the reason

**Comment by irisdingbj** [comment]
2024-02-07T03:38:59.937210Z  WARN inbound{^[[3mid^[[0m^[[2m=^[[0mda468de6f15cfd1539e6af8a9f7c707a ^[[3mpeer_ip^[[0m^[[2m=^[[0m10.244.138.36 ^[[3mpeer_id^[[0m^[[2m=^[[0mspiffe://cluster.local/ns/bookinfo-1-39307/sa/bookinfo-productpage}: ztunnel::proxy::inbound: connection to 10.244.138.16:9080 failed: Connection refused (os error 111) dur=112.7µs
2024-02-07T03:38:59.937356Z  WARN proxy{^[[3muid^[[0m^[[2m=^[[0m0cbc3388-4ec7-4c35-b963-67e83bb82e12}:outbound{^[[3mid^[[0m^[[2m=^[[0mda468de6f15cfd1539e6af8a9f7c707a}: ztunnel::proxy::outbound: failed dur=2.932183ms err=http status: 503 Service Unavailable

**Comment by bleggett** [comment]
> 2024-02-07T03:38:59.937210Z WARN inbound{^[[3mid^[[0m^[[2m=^[[0mda468de6f15cfd1539e6af8a9f7c707a ^[[3mpeer_ip^[[0m^[[2m=^[[0m10.244.138.36 ^[[3mpeer_id^[[0m^[[2m=^[[0mspiffe://cluster.local/ns/bookinfo-1-39307/sa/bookinfo-productpage}: ztunnel::proxy::inbound: connection to 10.244.138.16:9080 failed: Connection refused (os error 111) dur=112.7µs 2024-02-07T03:38:59.937356Z WARN proxy{^[[3muid^[[0m^[[2m=^[[0m0cbc3388-4ec7-4c35-b963-67e83bb82e12}:outbound{^[[3mid^[[0m^[[2m=^[[0mda468de6f15cfd1539e6af8a9f7c707a}: ztunnel::proxy::outbound: failed dur=2.932183ms err=http status: 503 Service Unavailable
> 
> 2024-02-07T03:38:59.628120Z info Running command (without lock): iptables-restore --noflush -v 2024-02-07T03:38:59.629943Z error Command error output: xtables other problem: line 2 failed 2024-02-07T03:38:59.629971Z error iptables failed to restore iptables rules: exit status 1 2024-02-07T03:38:59.630014Z error ambient failed to update POD inpod: bookinfo-1-39307/productpage-v1-675fc69cf-5px9n exit status 1 2024-02-07T03:38:59.630024Z error ambient failed to add pod to ztunnel: exit status 1
> 
> @bleggett in case you can share some lights for above error in ztunnel and istio-cni for bookinfo test, thanks!

Not sure - the primary issue in that log is that `iptables` is failing to create rules inside the pod. We saw that a few times early in our internal impl but I thought we had fixed it. I will take a look and see if I can repro and understand why iptables is failing to run in the pod.

**Comment by bleggett** [comment]
I have not had a chance to look at the failure on my end, but it might be worth comparing the test setup here to what @ilrudie is doing in https://github.com/istio/istio/pull/49255 -> if it works there, it should work here with bookinfo, and can probably triage what's different using that PR as a reference.

File: tests/integration/ambient/main_test.go (hunk 1 above)
Comment (yuval-k) [reply tests/integration/ambient/main_test.go:90]: i remember having to enabled it as some tests assume no CNI broke for me: https://github.com/istio/istio/pull/48253/commits/1b2cbdfd0e81fb2188652c9d789b16398ef1e1f9

**Comment by bleggett** [comment]
Also @saiskee just reminded me that @howardjohn had already opened https://github.com/istio/istio/issues/46682 to actually remove the use of bookinfo tests from the ambient suites in favor of directly using `echo` instances, which should be just as good for our purposes, and faster to deploy.

**Pull request closed without merging**